web3.js/src/transaction.ts

```diff
@@ -542,6 +543,21 @@ export class Transaction {
     this.signatures[index].signature = Buffer.from(signature);
   }
 
+  async getEstimatedFee(connection: Connection): Promise<RpcResponseAndContext<number>> {
+    const feeMessage = this.compileMessage();
+
+    if (!feeMessage) {
+      throw new Error('A message is required, but none was provided');
+    }
+
+    if (!connection) {
+      throw new Error('A connection is required, but none was provided.');
+    }
```

Comment:
`compileMessage` always returns a `Message`, `connection` is always a `Connection`. We have types for this, no need to do runtime checks!

Reply:
After talking with Jacob earlier I was planning to remove the message check. I added the connection check in the event a user fails to provide a connection as an argument when using this method. If they're developing with TypeScript, that won't be an issue but if they're just using JavaScript they'll get `TypeError: Cannot read properties of undefined (reading 'getFeeForMessage')`. I thought it would just make it obvious what needs to be fixed. That being said, I can definitely remove the connection check as well. Thanks for the feedback!

Reply:
Yeah, we shouldn't optimize for people using just JS. This is a TypeScript project. It adds runtime overhead and it makes the code duplicative.